sdk/test/AzureTest/AzureTestTests/AzureTestTests.swift

```diff
@@ -0,0 +1,167 @@
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import XCTest
+import Foundation
+import DVR
+import AzureTest
+
+class AzureTestTests: XCTestCase {
+
+    var fakeData: Data!
+    
+    var fakeRequest: URLRequest!
+    
+    var fakeResponse: URLResponse!
+    
+    var fakeResponseData: Data!
+    
+    private func chooseRecordedInteraction(number: Int) {
+        fakeRequest = fakeData.request(number: number)
+        fakeResponse = fakeData.response(number: number)
+        fakeResponseData = fakeData.responseData(number: number)
+    }
+    
+    override func setUpWithError() throws {
+        let testBundle = Bundle(for: type(of: self))
+        let path = testBundle.path(forResource: "TestData", ofType: "json")
+        fakeData = try! Data(contentsOf: URL(fileURLWithPath: path!))
+        chooseRecordedInteraction(number: 8)
+        
+    }
+
+    func test_scrubbingRequest_removeSubscriptionIDs() throws {
+        let cleanedRequest = Filter().scrubSubscriptionIDs(from: fakeRequest!)
+        let shouldPass = cleanedRequest.url?.absoluteString.contains(regex: Filter.subcriptionIDReplacement) ?? false
+        XCTAssert(shouldPass)
+        
+    }
+
+    func test_scrubbingResponse_removeSubscriptionIDs() throws {
+        let dirtyHeaders = ["location": "[\"https://management.azure.com/subscriptions/72f988bf-86f1-41af-91ab-2d7cd011db47/providers/Microsoft.KeyVault/locations/eastus/operationResults/VVR8MDYzNzU0NDA3MTY0MzE2NTczMnwwNjZENTEwRTA4N0U0MTY5ODc1MDhDRDY3QUJDMzdGOQ?api-version=2019-09-01\"]"]
+        
+        let dirtyBody = """
+            "string": "{\"id\":\"/subscriptions/72f988bf-86f1-41af-91ab-2d7cd011db47/providers/Microsoft.KeyVault/locations/eastus/deletedVaults/myValtZikfikxz\",\"name\":\"myValtZikfikxz\",\"type\":\"Microsoft.KeyVault/deletedVaults\",\"properties\":{\"vaultId\":\"/subscriptions/72f988bf-86f1-41af-91ab-2d7cd011db47/resourceGroups/rgname/providers/Microsoft.KeyVault/vaults/myValtZikfikxz\",\"location\":\"eastus\",\"tags\":{},\"deletionDate\":\"2021-04-19T05:32:42Z\",\"scheduledPurgeDate\":\"2021-07-18T05:32:42Z\"}}"
+        """
+        
+        let cleanLocation = Filter.scrubSubscriptionId(from: dirtyHeaders["location"]!)
+        let cleanBody = Filter.scrubSubscriptionId(from: dirtyBody)
+        
+        let shouldPass = !(cleanLocation.contains(regex: "72f988bf-86f1-41af-91ab-2d7cd011db47")) && !(cleanBody.contains(regex: "72f988bf-86f1-41af-91ab-2d7cd011db47"))
```

Comment:
Perhaps extract out the GUID to a constant. More maintainable via the DRY principle in case you every need to change it (for whatever reason).

Reply:
Should I just remove the test json and only take the pieces that we want to use and hard code them in? We aren't using much of that json file at all.

Reply:
Yeah, that's probably fine.